### PR TITLE
MM-52337: Use aws_opensearch_domain

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
We decided to use the latest OpenSearch version compatible with ElasticSearch (i.e., those until the latest OSS version of ElasticSearch, v7.10).

This PR addresses this decision, replacing the aws_elasticsearch_domain resource with aws_opensearch_domain. This forced me to upgrade the Terraform's aws provider itself, and some things had to be changed because of that.

This has been tested in a deployment using an `r6g.large.search` instance with  `Elasticsearch_7.10`. It deployed successfully, and a manual index of the 20M posts DB finished in 5 hours.

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52337

### CODE REVIEW

File: deployment/terraform/assets/.terraform.lock.hcl [hunk 1]
```diff
@@ -2,21 +2,81 @@
 # Manual edits may be lost in future updates.
```
Comment (agnivade) [inline deployment/terraform/assets/.terraform.lock.hcl:2]: Is this file meant to be checked in?

**Review by agnivade — APPROVED** [review]

File: deployment/terraform/assets/.terraform.lock.hcl (hunk 1 above)
Comment (agarciamontoro) [reply deployment/terraform/assets/.terraform.lock.hcl:2]: image

You asked this exact same question almost [three years ago](https://github.com/mattermost/mattermost-load-test-ng/pull/457#discussion_r634477970) hahaha

Answering your question: yes. I had the same question you had three years ago (and today), double-checked the docs and yes, they still recommend to check it in.

Comment (agnivade) [reply deployment/terraform/assets/.terraform.lock.hcl:2]: :facepalm:

**Review by isacikgoz — APPROVED** [review]
LGTM, love to see `defaults` package getting more and more features :)

**Comment by agarciamontoro** [comment]
> love to see defaults package getting more and more features

I'm always impressed about the groundwork of the `defaults.Validate` function, it's so easy to add a new validation! A very nice piece of code indeed :)

**Comment by agarciamontoro** [comment]
Merging, this has already been successfully tested.

**Pull request merged**